File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/VaultApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.Proof
import my.ssdid.sdk.domain.vault.Vault
import kotlinx.serialization.json.JsonObject

class VaultApi internal constructor(private val vault: Vault) {
    suspend fun sign(keyId: String, data: ByteArray): Result<ByteArray> = vault.sign(keyId, data)
    suspend fun createProof(
        keyId: String,
        document: JsonObject,
        proofPurpose: String,
        challenge: String? = null,
        domain: String? = null
    ): Result<Proof> = vault.createProof(keyId, document, proofPurpose, challenge, domain)
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/VaultApi.swift
import Foundation

/// Facade for cryptographic vault operations (signing, proof creation).
public struct VaultApi {
    let vault: Vault

    public func sign(keyId: String, data: Data) async throws -> Data {
        try await vault.sign(keyId: keyId, data: data)
    }

    public func createProof(
        keyId: String,
        document: [String: Any],
        proofPurpose: String,
        challenge: String? = nil,
        domain: String? = nil
    ) async throws -> Proof {
        try await vault.createProof(
            keyId: keyId,
            document: document,
            proofPurpose: proofPurpose,
            challenge: challenge,
            domain: domain
        )
    }
}
